services/backend/src/core/geds/geds.js

```diff
@@ -43,54 +43,284 @@ async function getGedsAssist(request, response) {
+
+      const enAddr = branchOrg.addressInformation.address.en;
+
       try {
         const location = await prisma.opOfficeLocation.findMany({
           where: {
-            country: branchOrg.addressInformation.country,
-            city: branchOrg.addressInformation.city,
-            postalCode: branchOrg.addressInformation.postalCode,
-            streetNumber: branchOrg.addressInformation.streetNumber,
-            translations: {
-              some: {
-                province: branchOrg.addressInformation.province,
-                streetName: branchOrg.addressInformation.streetNumber,
-              },
-            },
+            country: branchOrg.addressInformation.country.en,
+            city: branchOrg.addressInformation.city.en,
+            postalCode: branchOrg.addressInformation.pc,
+            streetNumber: parseInt(enAddr.split(" ")[0], 10),
           },
         });
 
         const profile = {
           firstName: dataGEDS.givenName,
           lastName: dataGEDS.surname,
-          jobTitle: {
-            en: dataGEDS.title.en,
-            fr: dataGEDS.title.fr,
-          },
+
+          locationId: location[0].id,
           email: dataGEDS.contactInformation.email,
+          branch: {
+            ENGLISH: branchOrg.description.en,
+            FRENCH: branchOrg.description.en,
+          },
           telephone: dataGEDS.phoneNumber,
           cellphone: dataGEDS.altPhoneNumber,
-          branchEn: branchOrg.description.en,
-          branchFr: branchOrg.description.fr,
-          organizations: organizations,
-          location: {
-            id: location.id,
-            description: {
-              en: `${location[0].addressEn}, ${location[0].city}, ${location[0].provinceEn}`,
-              fr: `${location[0].addressFr}, ${location[0].city}, ${location[0].provinceFr}`,
+          jobTitle: {
+            ENGLISH: dataGEDS.title.en,
+            FRENCH: dataGEDS.title.fr,
+          },
+          organizations: [
+            organizations.map((org) => ({
+              title: {
+                ENGLISH: org.description.en,
+                FRENCH: org.description.fr,
+              },
+              id: org.id,
+              tier: org.tier,
+            })),
+          ],
+        };
+
+        response.status(200).send(profile);
+      } catch (error) {
+        console.log(error);
+        response.status(500).json("Unable to parse geds data");
+      }
+    })
+    .catch((error) => {
+      console.log(error);
+      response.status(500).json("Unable to get geds Profile");
+    });
+}
+
+async function getGedsSync(request, response) {
+  const { id } = request.params;
+  const { name } = request.query;
+  const nameArray = name.split(" ");
+
+  const url = `${process.env.GEDSAPIURL}employees?searchValue=${nameArray[1]}%2C%20${nameArray[0]}&searchField=0&searchCriterion=2&searchScope=sub&searchFilter=2&maxEntries=200&pageNumber=1&returnOrganizationInformation=yes`;
+
+  const promises = [
```

Comment:
Just a small comment here add a time out for the API call. We are using 5 seconds.

Reply:
Added timeout